lib/index.coffee

```diff
@@ -51,6 +53,29 @@ streamToString = (stream) ->
 		.on 'end', ->
 			resolve(Buffer.concat(chunks).toString())
 
+setupMetrics = (metrics) ->
+	metrics.describe('gc_space_reclaimed', 'Disk space the GC must free')
+	metrics.describe('gc_number_images_to_remove', 'Number of images to be removed at a given time')
+	metrics.describe('gc_duration_milliseconds', 'Milliseconds taken to free GC the reclaimed space')
+	metrics.describe('gc_image_removal_errors',
+		'Number of image removal errors by type',
+		{
+			buckets: [ 404, 409, 500 ]
+		}
+	)
+
+	metrics.describe(
+		'gc_images_removed',
+		'Total number of images removed by all different methods',
+		{
+			buckets: ['tag', 'digest', 'id']
+		}
+	)
+
+recordGcRunTime = (metrics, startTime) ->
+	duration = Date.now() - startTime
```

Comment:
This should be timed using:

```coffeescript
startTime = process.hrtime();
```

and then this function would be:

```coffeescript
recordGcRunTime = (metrics, t0) ->
    t1 = process.hrtime();
    duration = t1[0] - t0[0] + (t1[1] - t0[1]) / 1e6;
    ....
```

Reply:
:+1: